docs/rules/compound-button-needs-labelling-v9.md

```diff
@@ -0,0 +1,39 @@
+# Accessibility: Compound buttons must have accessible labelling: title, aria-label, aria-labelledby, aria-describedby (`@microsoft/fluentui-jsx-a11y/compound-button-needs-labelling-v9`)
+
+<!-- end auto-generated rule header -->
+
+All interactive elements must have an accessible name.
+Please add title, aria-label, aria-labelledby, aria-describedBy, secondaryContent etc.
+
+<https://www.w3.org/TR/html-aria/>
+
+## Rule Details
+
+This rule aims to prevent an compound button from not having an accessible name.
+
+Examples of **incorrect** code for this rule:
+
+```jsx
+<CompoundButton/>
+<CompoundButton></CompoundButton>
+
+<Label id="calendar-2">Start date</Label>
+<CompoundButton aria-labelledby="calendar" />
+```
+
+Examples of **correct** code for this rule:
+
+```jsx
+<CompoundButton title="Compound example" />
```

Comment:
Without an icon, these buttons are actually not accessible for sighted users. We can suggest in the documentation:
```

 <CompoundButton>
    Example
  </CompoundButton>
  
  <CompoundButton secondaryContent="Secondary content">
    Example
  </CompoundButton>
  
   <CompoundButton
    icon={<CalendarMonthRegular />}
    secondaryContent="Secondary content"
  >
    Example
  </CompoundButton>
   
       <Tooltip content="With calendar icon only" relationship="label">
        <CompoundButton icon={<CalendarMonthRegular />} />
      </Tooltip>
	  
  <CompoundButton
    icon={<CalendarMonthRegular />}
    aria-label="Monthly calendar"
  >
  </CompoundButton>
	  
  
  <CompoundButton
    icon={<CalendarMonthRegular />}
    secondaryContent="Secondary content"
  >
  </CompoundButton>
  
  
  <CompoundButton
    icon={<CalendarMonthRegular />}
    secondaryContent="Secondary content"
  >
  </CompoundButton>
  
  
    <Label id="calendar-1">Compound example</Label>
    <CompoundButton aria-labelledby="calendar-1" icon={<CalendarMonthRegular />} />
```

  incorrect
  
```
    <CompoundButton></CompoundButton>
  
  
    <CompoundButton icon={<CalendarMonthRegular />}></CompoundButton>
	
	
	<Label>Compound example</Label>
    <CompoundButton icon={<CalendarMonthRegular />} />
```